treessence/src/test/java/fr/bipi/treessence/file/FileLoggerTreeTest.kt

```diff
@@ -0,0 +1,396 @@
+package fr.bipi.treessence.file
+
+import android.content.Context
+import android.util.Log
+import fr.bipi.treessence.common.formatter.LogcatFormatter
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.unmockkAll
+import junit.framework.TestCase
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import org.amshove.kluent.internal.assertEquals
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.io.TempDir
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import timber.log.Timber
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileReader
+import java.nio.file.Path
+import java.util.concurrent.atomic.AtomicInteger
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(Parameterized::class)
+class FileLoggerTreeTest(
+    private val coroutineScope: CoroutineScope?
+) {
+
+    @TempDir
+    private val tempDir: Path? = null
+
+    @MockK
+    private lateinit var context: Context
+
+    @MockK
+    private lateinit var file: File
+
+    @MockK
+    private lateinit var logcatFormatter: LogcatFormatter
+
+    private val logId = AtomicInteger(0)
+    private lateinit var dir: String
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this, relaxUnitFun = true)
+        dir = tempDir?.toAbsolutePath().toString()
+        setupDefaultMocks()
+        Timber.uprootAll()
+    }
+
+    @After
+    fun teardown() {
+        unmockkAll()
+        File(dir).deleteRecursively()
+        Timber.uprootAll()
+    }
+
+    @Test
+    fun testBasicFileLogging() = runTest {
+        val dirPath = "$dir/ce_basic_test_logs"
+        val fileLoggerTree = buildTree(1000000, dirPath)
+
+        Timber.plant(fileLoggerTree)
+
+        Timber.v("Verbose message") // Checks if priority is respected: Priority is DEBUG
+        Timber.d("Debug message")
+        Timber.i("Info message")
+        Timber.w("Warning message")
+        Timber.e("Error message")
+        Timber.wtf("Assert message")
+
+        val filename = fileLoggerTree.getFileName(0)
+        MatcherAssert.assertThat(filename, CoreMatchers.`is`("$dirPath/log.0"))
```

Comment:
Please import `assertThat` and `is` so that this can read:
```suggestion
        assertThat(filename, `is`("$dirPath/log.0"))
```